src/ray/common/status.h

```diff
@@ -101,6 +101,12 @@ enum class StatusCode : char {
   ObjectAlreadySealed = 23,
   ObjectStoreFull = 24,
   TransientObjectStoreFull = 25,
+  // worker pool status
```

Comment:
Hey, can we actually not put these as part of the common StatusCodes? Let's make these an enum member of the WorkerPool class instead. These error codes are pretty specific to the worker pool logic. We can change the callback signature to return this enum instead of a Status.

In general, we should reserve the common StatusCodes for cross-process communication only.

Reply:
Good point. I added an enum `PopWorkerStatus`.